pages/k8s/cni-sriov.md

```diff
@@ -0,0 +1,233 @@
+---
+wrapper_template: "kubernetes/docs/base_docs.html"
+markdown_includes:
+  nav: "kubernetes/docs/shared/_side-navigation.md"
+context:
+  title: "CNI with SR-IOV"
+  description: How to deploy and use SR-IOV CNI in Charmed Kubernetes
+keywords: CNI, networking, sr-iov, sriov
+tags: [operating]
+sidebar: k8smain-sidebar
+permalink: cni-sriov.html
+layout: [base, ubuntu-com]
+toc: False
+---
+
+SR-IOV support in **Charmed Kubernetes** makes it possible to take network
+interfaces that are SR-IOV Virtual Functions (VFs) and attach them directly to
+pods. SR-IOV support is provided by the sriov-cni and
+sriov-network-device-plugin charms, which must be deployed into a Kubernetes
+model in Juju.
+
+## Requirements
+
+### Juju 2.8.0
+
+The SR-IOV CNI charms require Juju 2.8.0 or newer.
+
+### SR-IOV Hardware Support
+
+The kubernetes-worker host machines in your **Charmed Kubernetes** cluster must
+have attached network interface cards that support SR-IOV. While you will be
+attaching Virtual Functions to pods, the Physical Functions **must** be present
+on the hosts as well.
+
+### CNI providers
+
+SR-IOV CNI is not a replacement for other CNI providers. Your
+**Charmed Kubernetes** deployment must include at least one of the base CNI
+providers documented in the [CNI overview][cni-overview] page.
+
+### Multus
+
+SR-IOV CNI requires Multus. You can read about how to enable Multus on
+**Charmed Kubernetes** in the [Multus CNI][cni-multus] page.
+
+### Persistent volume support
+
+In order to deploy SR-IOV CNI, you will need a Kubernetes model in Juju, which
+requires persistent volume support to be enabled in your **Charmed Kubernetes**
+cluster.
+
+If your cluster includes any of the cloud integrator charms, then you should
+have persistent volume support already. Otherwise, you can read the
+[Storage][storage] documentation page to learn how to enable persistent volume
+support by adding Ceph to your cluster.
+
+### Creating a Kubernetes model in Juju
```

Comment:
My first thought is that this might belong in the Operations/Addons page of the documentation, since Kubernetes charms are where we intend to go long-term with all addons, not just Multus/SR-IOV. What do you think?

Reply:
Yes, I agree. I think for the purposes of getting 1.19 out we should probably go ahead with the duplication for now, and make a new page and PR for de-duplicating, which isn't as urgent.